ops/c/src/mpi/ops_mpi_partition.cpp

```diff
@@ -171,8 +171,8 @@ void ops_partition_blocks(int **processes, int **proc_offsets, int **proc_disps,
       //Sanity check for force decomp
       int prod = 1;
       for (int d = 0; d < ndim; d++) prod *= (pdims[d]==0 ? 1 : pdims[d]);
-      if (prod > processes_per_block[i]) throw OPSException(OPS_RUNTIME_CONFIGURATION_ERROR, "Error: force_decomp requested more processes than available for the block");
-      
+      if (prod != processes_per_block[i]) throw OPSException(OPS_RUNTIME_CONFIGURATION_ERROR, "Error: force_decomp requested processes inconsistent with one specified for the block");
```

Comment:
That is not right - you can specify 0 if you don't care about the number of processes in that direction. So prod needs to be less or equal, so MPI_Dims_create will work.

Reply:
Ok, See. Can we only pick up the right one here?

Reply:
I'm not sure what you mean. My original check is good for the case where some decomp sizes are not specified

Reply:
I mean if we can pick up your comment to merge and ignore my wrong commit by using Github?

Reply:
I reverted it. please see blow.